common/src/main/cpp/extensions/openxr_fb_passthrough_extension_wrapper.cpp

```diff
@@ -607,6 +639,137 @@ bool OpenXRFbPassthroughExtensionWrapper::has_layer_depth_passthrough_capability
+	target_color_lut_handle = p_target_color_lut->get_handle();
+
+	current_passthrough_filter = PASSTHROUGH_FILTER_COLOR_MAP_INTERPOLATED_LUT;
+	color_map_interpolated_lut.sourceColorLut = source_color_lut_handle;
+	color_map_interpolated_lut.targetColorLut = target_color_lut_handle;
+	color_map_interpolated_lut.weight = CLAMP(p_weight, 0.0, 1.0);
+	passthrough_style.next = &color_map_interpolated_lut;
+
+	if (is_passthrough_started()) {
+		XrResult result = xrPassthroughLayerSetStyleFB(passthrough_layer[current_passthrough_layer], &passthrough_style);
+		if (XR_FAILED(result)) {
+			UtilityFunctions::print("Failed to set passthrough style, error code: ", result);
+		}
+	}
+}
+
+void OpenXRFbPassthroughExtensionWrapper::create_color_lut(const Ref<OpenXRMetaPassthroughColorLut> &p_color_lut) {
+	if (p_color_lut->get_image_cell_resolution() > system_passthrough_color_lut_properties.maxColorLutResolution) {
+		UtilityFunctions::print("Color LUT cell resolution cannot be greater than the maximum resolution supported by this system: ", system_passthrough_color_lut_properties.maxColorLutResolution);
+		return;
+	}
+
+	XrPassthroughColorLutChannelsMETA channels;
+	switch (p_color_lut->get_channels()) {
+		case OpenXRMetaPassthroughColorLut::COLOR_LUT_CHANNELS_RGB: {
+			channels = XR_PASSTHROUGH_COLOR_LUT_CHANNELS_RGB_META;
+		} break;
+		case OpenXRMetaPassthroughColorLut::COLOR_LUT_CHANNELS_RGBA: {
+			channels = XR_PASSTHROUGH_COLOR_LUT_CHANNELS_RGBA_META;
+		} break;
+	}
+
+	XrPassthroughColorLutDataMETA color_lut_data = {
+		(uint32_t)p_color_lut->get_buffer().size(), // bufferSize
+		p_color_lut->get_buffer().ptr(), // buffer
+	};
+
+	XrPassthroughColorLutCreateInfoMETA color_lut_create_info = {
+		XR_TYPE_PASSTHROUGH_COLOR_LUT_CREATE_INFO_META, // type
+		nullptr, // next
+		channels, // channels
+		(uint32_t)p_color_lut->get_image_cell_resolution(), // resolution
+		color_lut_data, // data
+	};
+
+	XrPassthroughColorLutMETA handle;
+	XrResult result = xrCreatePassthroughColorLutMETA(passthrough_handle, &color_lut_create_info, &handle);
+	if (XR_FAILED(result)) {
+		UtilityFunctions::print("Failed to create passthrough color LUT, error code: ", result);
+		return;
+	}
+
+	p_color_lut->set_handle(handle);
+}
+
+void OpenXRFbPassthroughExtensionWrapper::destroy_color_lut(const Ref<OpenXRMetaPassthroughColorLut> &p_color_lut) {
+	XrPassthroughColorLutMETA handle = p_color_lut->get_handle();
+
+	if (handle == XR_NULL_HANDLE) {
+		UtilityFunctions::print("Cannot delete invalid color LUT");
+		return;
+	}
+
+	XrResult result = xrDestroyPassthroughColorLutMETA(handle);
+	if (XR_FAILED(result)) {
+		UtilityFunctions::print("Failed to destroy passthrough color LUT, error code: ", result);
+		return;
+	}
+
+	if (color_lut_handle == handle) {
+		color_lut_handle = XR_NULL_HANDLE;
+	}
+	if (source_color_lut_handle == handle) {
+		color_lut_handle = XR_NULL_HANDLE;
+	}
+	if (target_color_lut_handle == handle) {
+		color_lut_handle = XR_NULL_HANDLE;
+	}
```

Comment:
Do you mean for those to be `source_color_lut_handle` and `target_color_lut_handle` instead of `color_lut_handle`?

Reply:
Yes! Fixed in latest push.